src/onboarding/steps.ts

```diff
@@ -0,0 +1,214 @@
+  const supportedBiometryType = supportedBiometryTypeSelector(state)
+  const skipVerification = skipVerificationSelector(state)
+  const numberAlreadyVerifiedCentrally = numberVerifiedCentrallySelector(state)
+
+  return {
+    recoveringFromStoreWipe,
+    choseToRestoreAccount,
+    supportedBiometryType,
+    skipVerification,
+    numberAlreadyVerifiedCentrally,
+  }
+}
+
+/**
+ * Traverses through the directed graph of onboarding navigate, navigateClearingStack, and navigateHome calls
+ * and count the number of screens until the given screen and the total number
+ */
+export function getOnboardingStepValues(screen: Screens, onboardingProps: OnboardingProps) {
+  let stepCounter = 1 // will increment this up to the onboarding step the user is on
+  let totalCounter = 1
+  let reachedScreeen = false // tracks whether we have reached the screen the user is on in onboarding, and we can stop incrementing stepCounter
+  let currentScreen: Screens = FIRST_ONBOARDING_SCREEN // pointer that we will update when simulating navigation through the onboarding screens to calculate "step" and "totalSteps"
+
+  const navigate = (s: Screens) => {
+    // dummy navigation function to help determine what onboarding step the user is on, without triggering side effects like actually cycling them back through the first few onboarding screens
+    totalCounter++
+    if (currentScreen === screen) {
+      reachedScreeen = true
+    }
+    if (!reachedScreeen) {
+      stepCounter++
+    }
+    currentScreen = s
+  }
+
+  const navigateHome = () => {
+    currentScreen = Screens.WalletHome
+  }
+
+  // @ts-ignore: Compiler doesn't understand that navigate() can update currentScreen
+  while (currentScreen !== END_OF_ONBOARDING_SCREEN) {
+    const stepInfo = _getStepInfo({
+      firstScreenInStep: currentScreen,
+      navigator: {
+        navigate,
+        popToScreen: () => {
+          // no-op
+        },
+        navigateHome,
+        navigateClearingStack: navigate,
+      },
+      dispatch: () => {
+        // no-op
+      },
+      props: onboardingProps,
+    })
+    if (!stepInfo) {
+      throw new Error(`No step info found for ${currentScreen}.`)
+    }
+    stepInfo?.next()
+  }
+
+  return {
+    totalSteps: totalCounter,
+    step: stepCounter,
+  }
+}
+
+export function goToNextOnboardingScreen({
+  firstScreenInCurrentStep,
+  onboardingProps,
+}: {
+  firstScreenInCurrentStep: Screens
+  onboardingProps: OnboardingProps
+}) {
+  const stepInfo = _getStepInfo({
+    firstScreenInStep: firstScreenInCurrentStep,
+    navigator: {
+      navigate: NavigationService.navigate,
+      popToScreen: NavigationService.popToScreen,
+      navigateHome: NavigationService.navigateHome,
+      navigateClearingStack: NavigationService.navigateClearingStack,
+    },
+    dispatch: store.dispatch,
+    props: onboardingProps,
+  })
+  stepInfo?.next()
+}
+
+export function _getStepInfo({ firstScreenInStep, navigator, dispatch, props }: GetStepInfoProps) {
```

Comment:
I think a docstring here could be helpful, since this is now the source of truth for onboarding navigation logic. Probably would be useful to mention this.